src/nifgen/system_tests/test_system_nifgen.py

```diff
@@ -50,3 +50,70 @@ def test_get_self_cal_last_date_and_time():
 
 def test_self_cal(session):
         session.self_cal()
+
+
+def test_create_waveform(session):
+    waveform_data = [0.000000, 0.049068, 0.098017, 0.146730, 0.195090, 0.242980, 0.290285, 0.336890, 0.382683, 0.427555]
+    session.output_mode = nifgen.OutputMode.NIFGEN_VAL_OUTPUT_ARB  # TODO(Jaleel): name to change per #553
+    session.configure_arb_waveform(session.create_waveform_f64(waveform_data), 1.0, 0.0)
+
+
+def test_disable(session):
+    channel = session['0']
+    assert channel.output_enabled is True
+    session.disable()
+    assert channel.output_enabled is False
+
+
+def test_get_ext_cal_last_date_and_time():
+    with nifgen.Session('', False, 'Simulate=1, DriverSetup=Model:5421;BoardType:PXI') as session:  # 5433 throws out unrecoverable error on calling get_ext_cal_last_date_and_time()
+        try:
+            session.get_ext_cal_last_date_and_time()
+            assert False
+        except nifgen.Error as e:
+            assert e.code == -1074118632  # This operation is not supported for simulated device
+
+
+def test_get_ext_cal_last_temp(session):
+    try:
+        session.get_ext_cal_last_temp()
+    except nifgen.Error as e:
+        assert e.code == -1074135023  # Function or method not supported for 5413/23/33 and not supported on any other FGen when simulated
+
+
+def test_get_ext_cal_recommended_interval(session):
+    recommended_interval = session.get_ext_cal_recommended_interval()
+    assert recommended_interval == 24  # recommended external cal interval is 24 months
+
+
+''' TODO(Jaleel) Enable after Issue#558 fixed
+def test_get_hardware_state():
+    with nifgen.Session('', False, 'Simulate=1, DriverSetup=Model:5421;BoardType:PXI') as session:  # Function or method not supported for 5413/23/33
+        assert session.get_hardware_state() == nifgen.HardwareState.NIFGEN_VAL_IDLE  # TODO(Jaleel): name to change per #553
```

Comment:
I think you should leave this test enabled and look for the int value. That way when the issue is fixed we cannot forget to re-enable this, it will simply break.

Reply:
Calling this function itself will throw the error right ? According to session.py test_get_hardware_state() is expected to return HardwareState which is not at all present .

Reply:
I see, it is in the function metadata but the enum doesn't exist in enums.py.
Ok. Please add the comment in #558 to remember to re-enable this.

Reply:
updated #558 and #538 with comments.